### PR TITLE
Update build tools including Go 1.16.2

### PR DESCRIPTION
**Pull request by bboreham**
- Stop creating a NodeJS build image, just use `node:10.19` directly.
 - Update to Go 1.16.2.
 - Remove some unused tools.
 - Fix some build issues on MacOS.
 - Add 'osusergo' tag to remove some link-time warnings.
 - Remove 'demo.json' file, not used in Scope.
 - Miscelaneous other build improvements.

### CODE REVIEW

File: backend/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.14.2
+FROM golang:1.16.2
```
Comment (dholbach) [inline backend/Dockerfile:1]: Will `go.mod` need an update too?

**Comment by dholbach** [comment]
Great work - this looks much cleaner now!

File: backend/Dockerfile (hunk 1 above)
Comment (bboreham) [reply backend/Dockerfile:1]: You mean the line in `go.mod` saying `go 1.13` ?
I read up on that https://golang.org/ref/mod#go-mod-file-go
It seems there would be some benefit from updating it.

If you mean updating the other Go package dependencies, then I'm happy to leave that to another PR.

Comment (dholbach) [reply backend/Dockerfile:1]: :+1:

**Pull request merged**